### DISCUSSION TITLE
Cross-section is missing a polygon when using Python bindings

### DISCUSSION THREAD

**Discussion by johnnybigert** [elalish/manifold#458] [Q&A] [answered]
Hi all,

I would like to ask for your help in an investigation that I've started. :)

I was running a Python program using the Python bindings for CrossSection and noticed that there was a discrepancy between the output of CrossSection and the expected output. I boiled it down to a small example and implemented it both in Python and C++ (see pull request https://github.com/elalish/manifold/pull/457). The interesting thing is that CrossSection works as intended in C++, but fails using Python! I have looked through the bindings, but I have not been able to spot the problem yet. Any ideas what could be the problem?

Some illustrations (the square is just used as a test in the pull request to show that there's a part of the polygon missing):

This is the expected result (and the result in C++):
![Screenshot 2023-06-17 at 23 21 39](https://github.com/elalish/manifold/assets/17929792/6a31d417-bf3b-4cc0-ae6a-b7e02c83ae83)

This is the observed result in Python:
![Screenshot 2023-06-17 at 23 21 56](https://github.com/elalish/manifold/assets/17929792/c9ef41e4-d476-4c3b-bf7d-2aff8cd8fd50)

This is what the gltf looks like from the Python program:
![Screenshot 2023-06-17 at 23 04 05](https://github.com/elalish/manifold/assets/17929792/1cae74a2-9b89-4c2c-8d64-f2678dc1f79c)

**Comment by elalish** [comment]
They both look wrong to me - shouldn't the top one have the rectangle as a hole? Do they have proper CCW orientation?

**Reply by johnnybigert** [reply]
Hi! No, the rectangle is not a hole, it's just a (separate) test rectangle that I used to show that there's no overlap when there should be.

The problematic polygon has CCW orientation, yes. So does the test rectangle.

**Comment by elalish** [comment, marked as answer]
In case anyone comes here wondering, we did eventually get to the bottom of this in #457 and it was fixed in subsequent PRs.
